### ISSUE TITLE
Lambda types not inferred, allow unsound code to pass

### ISSUE THREAD

**Issue by dphilipson** [microsoft/pyright#267]
Hi Pyright, love the work you're doing!

The Mypy docs have the [following example](https://mypy.readthedocs.io/en/stable/kinds_of_types.html?highlight=lambda#callable-types-and-lambdas):

```python
l = map(lambda x: x + 1, [1, 2, 3])   # Infer x as int and l as List[int]
```

However, in Pyright with the same example, `x` and `l` are inferred as `Unknown` and `Iterator[Unknown]` respectively. It would be great to get the same inference as in Mypy.

Along similar lines, the following typechecks, but fails at runtime:
```python
map(lambda x: len(x), [1, 2, 3])
# At runtime: TypeError: object of type 'int' has no len()
```

This using the VSCode plugin, version 1.0.66.

**Comment by erictraut** [comment]
Good suggestion. I'll look into it.

**Comment by erictraut** [comment]
This functionality will be in the next published version of pyright.

**Comment by erictraut** [comment]
This is now implemented in 1.0.70, which I just published.
